Source/santad/SNTDriverManager.m

```diff
@@ -23,67 +23,97 @@
 
 @interface SNTDriverManager ()
 @property io_connect_t connection;
+@property(readwrite) BOOL connectionEstablished;
 @end
 
 @implementation SNTDriverManager
 
-static const int MAX_DELAY = 15;
-
 #pragma mark init/dealloc
 
 - (instancetype)init {
   self = [super init];
   if (self) {
-    kern_return_t kr;
-    io_service_t serviceObject;
-    CFDictionaryRef classToMatch;
-
-    if (!(classToMatch = IOServiceMatching(USERCLIENT_CLASS))) {
-      LOGD(@"Failed to create matching dictionary");
+    CFDictionaryRef classToMatch = IOServiceMatching(USERCLIENT_CLASS);
+    if (!classToMatch) {
+      LOGE(@"Failed to create matching dictionary");
       return nil;
     }
 
     // Attempt to load driver. It may already be running, so ignore any return value.
     KextManagerLoadKextWithIdentifier(CFSTR(USERCLIENT_ID), NULL);
 
-    // Locate driver. Wait for it if necessary.
-    int delay = 1;
-    do {
-      CFRetain(classToMatch);  // this ref is released by IOServiceGetMatchingService
-      serviceObject = IOServiceGetMatchingService(kIOMasterPortDefault, classToMatch);
+    // Wait for the driver to appear
+    [self waitForDriver:classToMatch];
+  }
+  return self;
+}
 
-      if (!serviceObject) {
-        LOGD(@"Waiting for Santa driver to become available");
-        sleep(delay);
-        if (delay < MAX_DELAY) delay *= 2;
-      }
-    } while (!serviceObject);
-    CFRelease(classToMatch);
+- (void)dealloc {
```

Comment:
Given SNTDriverManager is essentially a singleton, this should never be called. Add a note.

Reply:
We might treat SNTDriverManager like a singleton but it isn't implemented as one.